packages/spectacle/src/hooks/use-slides.tsx

```diff
@@ -49,7 +48,8 @@ export function useSlide(
   doesSlideHaveTemplate: boolean,
   userProvidedId?: SlideId
 ) {
-  const [slideId] = useState<SlideId>(userProvidedId || ulid);
+  const id = useId();
```

Comment:
`ulid` was thrashing on SSR, thus why Spectacle had hydration issues prior. Since any client components are not rendered on the sever (like the fullscreen button) we can guarantee the node tree will remain consistent on first pass and `useId()` is the safer choice.

Reply:
Do we want to remove ulid as a dependency, and also bump our react peer dep to 18 since we're using useId?

Reply:
Yeah! Done. Good call.